api/internal/target/kusttarget.go

```diff
@@ -381,7 +381,7 @@ func (kt *KustTarget) runValidators(ra *accumulator.ResAccumulator) error {
 		if err = kt.removeValidatedByLabel(newMap); err != nil {
 			return err
 		}
-		if err = orignal.ErrorIfNotEqualSets(newMap); err != nil {
+		if err = orignal.ErrorIfNotEqualLists(newMap); err != nil {
```

Comment:
In line384, we want to make sure that resMap is not modified after executing Transform method (line376).
If we still use [ErrorIfNotEqualSets](https://github.com/kubernetes-sigs/kustomize/blob/master/api/resmap/reswrangler.go#L306), we have a following error.

```go
--- FAIL: TestFnContainerValidator (3.50s)
    harness.go:100: validator shouldn't modify the resource map: nodes unequal:
         -- {"apiVersion":"apps/v1","kind":"Deployment","metadata":{"labels":{"app":"nginx"},"name":"nginx"},"spec":{"selector":{"matchLabels":{"app":"nginx"}},"template":{"metadata":{"labels":{"app":"nginx"}},"spec":{"containers":[{"image":"nginx","name":"nginx"}]}}}},
         -- {"apiVersion":"apps/v1","kind":"Deployment","metadata":{"labels":{"app":"nginx"},"name":"nginx"},"spec":{"selector":{"matchLabels":{"app":"nginx"}},"template":{"metadata":{"labels":{"app":"nginx"}},"spec":{"containers":[{"image":"nginx","name":"nginx"}]}}}}

        --
        &resource.Resource{RNode:yaml.RNode{fieldPath:[]string(nil), value:(*yaml.Node)(0xc000412000), Match:[]string(nil)}, refVarNames:[]string(nil)}
        ------
        &resource.Resource{RNode:yaml.RNode{fieldPath:[]string(nil), value:(*yaml.Node)(0xc000573860), Match:[]string(nil)}, refVarNames:[]string(nil)}
```

This is because ErrorIfNotEqualsSets compares resMap’s underlying RNodes by reflect.DeepEqual (https://github.com/kubernetes-sigs/kustomize/blob/master/api/resmap/reswrangler.go#L329).
As we compare original and newMap which are already deep-copied, a value field of RNode struct always points to different addresses.
To avoid this, we can use[ ErrorIfNotEqualsLists](https://github.com/kubernetes-sigs/kustomize/blob/master/api/resmap/reswrangler.go#L343) method which compares resMaps as strings.

Reply:
Thanks for explaining!
I understand why you need to change.